src/network/bpf/sysctl_monitor/sysctl-monitor.bpf.c

```diff
@@ -0,0 +1,92 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "vmlinux.h"
+
+#include <bpf/bpf_helpers.h>
+
+#include "sysctl-write-event.h"
+
+struct {
+        __uint(type, BPF_MAP_TYPE_RINGBUF);
+        __uint(max_entries, 256 * 1024);
+} written_sysctls SEC(".maps");
+
+static bool my_streq(const char *s1, const char *s2, int l)
+{
+        for (int i = 0; i < l; i++) {
+                if (s1[i] != s2[i])
+                        return false;
+                if (!s1[i])
```

Comment:
in our codebase we try to avoid checking non-bool and non-ptrs for non-zeroness with C's downgrade-to-bool logic. Instead, just do an explicit check, i.e. `!= 0` or even `!= '\0'`